crates/teloxide/src/dispatching/dispatcher.rs

```diff
@@ -495,6 +495,32 @@ impl<R, Err, Key> Dispatcher<R, Err, Key> {
     }
 }
 
+/// Extension methods for adding `tracing` to [`UpdateHandler`].
+///
+/// This trait is only available if the `tracing` feature is enabled.
+#[cfg(feature = "tracing")]
+pub trait UpdateHandlerTracingExt<E> {
+    /// Returns an `UpdateHandler` with tracing enabled.
+    fn with_tracing(self, f: fn(Arc<Update>) -> tracing::Span) -> Self;
+}
+
+#[cfg(feature = "tracing")]
+impl<E: 'static> UpdateHandlerTracingExt<E> for UpdateHandler<E> {
+    fn with_tracing(self, f: fn(Arc<Update>) -> tracing::Span) -> Self {
```

Comment:
I would also prefer the method to have `span` in the name, something like `with_tracing_span` or maybe even just `spanned` although maybe it's ambiguous.

Reply:
Great, changing it to `with_tracing_span`